File: solutions/go/space-age/1/space_age.go
package space

type Planet string

var earthSeconds float64 = 31557600

func Age(seconds float64, planet Planet) float64 {
	if planet == "Earth" {
		return seconds / earthSeconds
	} else if planet == "Mercury" {
		return seconds / (earthSeconds * 0.2408467)
	} else if planet == "Venus" {
		return seconds / (earthSeconds * 0.61519726)
	} else if planet == "Mars" {
		return seconds / (earthSeconds * 1.8808158)
	} else if planet == "Jupiter" {
		return seconds / (earthSeconds * 11.862615)
	} else if planet == "Saturn" {
		return seconds / (earthSeconds * 29.447498)
	} else if planet == "Uranus" {
		return seconds / (earthSeconds * 84.016846)
	} else if planet == "Neptune" {
		return seconds / (earthSeconds * 164.79132)
	}
	return -1
}

File: solutions/c/space-age/1/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds) {
    float earth_years;
    earth_years = seconds / 31557600.0;
    switch (planet) {
        case MERCURY:
            return earth_years / 0.2408467;
        case VENUS:
            return earth_years / 0.61519726;
        case EARTH:
            return earth_years / 1.0;
        case MARS:
            return earth_years / 1.8808158;
        case JUPITER:
            return earth_years / 11.862615;
        case SATURN:
            return earth_years / 29.447498;
        case URANUS:
            return earth_years / 84.016846;
        case NEPTUNE:
            return earth_years / 164.79132;
        default:
            return -1.0;
    }
    return 0;
}
